my-react-app/src/components/Buttons/Buttons.tsx

```diff
@@ -13,11 +19,40 @@ const Buttons = () => {
     "bg-purple-400",
     "bg-teal-400",
   ];
+
+  const layoutClass =
+    layout === "flex"
+      ? "flex flex-column flex-wrap"
+      : "grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3";
+  const [containerColor, setContainerColor] = useState("");
```

Comment:
state handling immer ganz oben in der componente machen 

grundsätzlich gilt hooks relative oben in der componente handlen und die conditionals für die comp nahe am jsx code